File: result/translate/chatgpt/java2python/python/LONGEST_SUBARRAY_COUNT_1S_ONE_COUNT_0S.py
def f_gold(arr, n):
    um = {}
    sum_val = 0
    max_len = 0
    for i in range(n):
        sum_val += -1 if arr[i] == 0 else 1
        if sum_val == 1:
            max_len = i + 1
        elif sum_val not in um:
            um[sum_val] = i
        if sum_val - 1 in um:
            if max_len < (i - um[sum_val - 1]):
                max_len = i - um[sum_val - 1]
    return max_len

File: result/translate/transcoder/python2java/java/LONGEST_SUBARRAY_COUNT_1S_ONE_COUNT_0S.java
public static int f_gold ( int [ ] arr , int n ) {
  HashMap < Integer , Integer > um = new HashMap < Integer , Integer > ( ) ;
  int sum = 0 ;
  int maxLen = 0 ;
  for ( int i = 0 ;
  i < n ;
  i ++ ) {
    if ( arr [ i ] == 0 ) {
      sum ++ ;
    }
    else {
      sum ++ ;
    }
    if ( ( sum == 1 ) && ( ( i - 1 ) < n ) ) {
      maxLen = i + 1 ;
    }
    else if ( ( sum != 0 ) && ( ( i - 1 ) < n ) ) {
      um . put ( sum , i ) ;
    }
    if ( ( ( sum - 1 ) < n ) && ( ( i - 1 ) < n ) ) {
      if ( ( maxLen < ( i - um . get ( sum - 1 ) ) ) && ( ( i - um . get ( sum - 1 ) ) < n ) ) {
        maxLen = i - um . get ( sum - 1 ) ;
      }
    }
  }
  return maxLen ;
}
